File: Backend/entity/stock.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Stock struct {

	gorm.Model
	Quantity       uint      `json:"quantity"`

	Price          float64   `json:"price"`

	DateIn         time.Time `json:"date_in"`

	ExpirationDate time.Time `json:"expiration_date"`

	ProductID      uint

	Product        Product `gorm:"foreignKey:ProductID"`

	SupplierID     uint

	Supplier       Supplier `gorm:"foreignKey:SupplierID"`

	EmployeeID     uint
	
	Employee       Employee `gorm:"foreignKey:EmployeeID"`

}

File: Frontend/src/interfaces/Stock.ts
export interface StockInterface {
    stock_id: number;
    category_id: number;      // รหัสประเภท
    product_code_id: string;   // รหัสสินค้า
    product_name: string;     // ชื่อสินค้า
    quantity: number;        // จำนวนสินค้า
    price: number;           // ราคา
    date_in: string;          // วันที่นำเข้า
    expiration_date: string;  // วันหมดอายุ
    supplier_id: number;      // รหัสผู้จัดจำหน่าย
    employee_id: number;      // รหัสพนักงาน
  }